File: carcassonne-master/src/side_connections.cc
#include "string_utility.h"
#include "side_connections.h"

namespace carcassonne
{

SideConnections::SideConnections()
{
  clockwise_ = false;
  across_ = false;
  counterclockwise_ = false;
  middle_ = false;	
  
}

SideConnections::SideConnections(const SideConnections& src)
{
	CopyFrom(src);
}

SideConnections& SideConnections::
	operator=(const SideConnections& rhs)
{
	if(this == &rhs) {
		return (*this);
	}
	
	CopyFrom(rhs);
	
	return (*this);
}

void SideConnections::
	CopyFrom(const SideConnections& src)
{
  clockwise_ = src.clockwise_;
  across_ = src.across_;
  counterclockwise_ = src.counterclockwise_;
  middle_ = src.middle_;
}

SideConnections* SideConnections::
	Clone() const
{
	return new SideConnections(*this);
}

bool SideConnections::clockwise() { return clockwise_; }
bool SideConnections::across() { return across_; }
bool SideConnections::counterclockwise() { return counterclockwise_; }
bool SideConnections::middle() { return middle_; }

void SideConnections::set_clockwise(bool clockwise)
{ 
	clockwise_ = clockwise; 

}

void SideConnections::set_across(bool across)
{
	across_ = across;
	
}

void SideConnections::set_counterclockwise(bool counterclockwise)
{
	counterclockwise_ = counterclockwise;
		
}

void SideConnections::set_middle(bool middle)
{
	middle_ = middle;
	
}

std::vector<std::string> SideConnections::ToStringVector() const
{
  std::vector<std::string> tmp;

  if(clockwise_) {
    tmp.push_back("clockwise");
  }

  if(across_) {
    tmp.push_back("across");
  }

  if(counterclockwise_) {
    tmp.push_back("counterclockwise");
  }

  return tmp;
}

std::string SideConnections::ToString() const
{
  return "connection(s) to " + commaize(ToStringVector());
}

SideConnections::~SideConnections()
{
}

}

File: js/SideConnections.js
module.exports= class SideConnections {
    constructor(src) {
        this.clockwise_ = false;
        this.across_ = false;
        this.counterclockwise_ = false;
        this.middle_ = false;
        if (src) {
            this.CopyFrom(src);
        }
    }

    clockwise() {
        return this.clockwise_
    };

    across() {
        return this.across_
    };

    counterclockwise() {
        return this.counterclockwise_
    };

    middle() {
        return this.middle_
    };

    set_clockwise(clockwise) {
        this.clockwise_ = clockwise
    };

    set_across(across) {
        this.across_ = across
    };

    set_counterclockwise(counterclockwise) {
        this.counterclockwise_ = counterclockwise
    };

    set_middle(middle) {
        this.middle_ = middle
    };

    ToStringVector() {
        var tmp = [];
        if (this.clockwise_) {
            tmp.push("clockwise");
        }

        if (this.across_) {
            tmp.push("across");
        }

        if (this.counterclockwise_) {
            tmp.push("counterclockwise");
        }

        return tmp;
    }

    ToString() {
        return "connection(s) to " + this.ToStringVector();
    }

    CopyFrom(src) {
        this.clockwise_ = src.clockwise_;
        this.across_ = src.across_;
        this.counterclockwise_ = src.counterclockwise_;
        this.middle_ = src.middle_;

    }

    Clone() {
        return new SideConnections(this);
    }
}